### ISSUE TITLE
Header tooltip not always displaying

### ISSUE THREAD

**Issue by jackdorsey** [JAM-Software/Virtual-TreeView#728]
[HeaderBug.zip](https://github.com/Virtual-TreeView/Virtual-TreeView/files/1273719/HeaderBug.zip)

Attached is a sample project - hover over the header - the tooltip is not always displayed.

**Comment by jackdorsey** [comment]
Actually the tooltip starts displaying properly only after you click on a header and then hover a column header that is not highlighted.

**Comment by joachimmarder** [comment]
A tricky one. In a few cases I see the describes behavior, but this is difficult to debug because it does not occur deterministic and stopping at a breakpoint changes the further behavior. I have no clue so far.

**Comment by joachimmarder** [comment]
> Actually the tooltip starts displaying properly only after you click on a header and then hover a column header that is not highlighted.

Even hovering was usually sufficient to get the tooltip at a later point.

**Comment by joachimmarder** [comment]
Sorry, but after several hours of debugging I don't think it is worth the effort. In my case the hint is missing only rarely and never after the first user interaction took place.

**Comment by jackdorsey** [comment]
Hello
I did not check if you changed something, but it seems a bit better than before. (I got the latest version to test your fix on the load header/repaint issue)

Further tests show that hovering the header from top or left - without entering the VT client area prior to entering the header cell almost never displays the tooltip. If you like you can check as this code part is still fresh in your head I guess. Current situation is not that bad, but I think this issue should stay open with low priority at least in case you decide not to investigate further.

Thank you.

**Comment by joachimmarder** [comment]
> I did not check if you changed something,

I did not change anything. 

> Further tests show that hovering the header from top or left - without entering the VT client area prior to entering the header cell almost never displays the tooltip.

I agree, the likelihood seems slightly higher that the issue occurs. There is however no code that is specific for that. I guess it makes a difference if the mouse hovered over the client area of the control before hovering over the header.

It would help if there would be defintive and longer lasting repro steps, which is why i flagged this issue accordingly. I will leave it open for some more months, but I doubt that something will change.

**Comment by IgitBuh** [comment]
I tried the demo project and noticed an additional detail ... or actually another problem. I don't know if it has the same cause or if they're unrelated. But I thought, it's worth pointing it out:
![tvt-header-tooltip](https://user-images.githubusercontent.com/703571/36640795-d19ff4b6-1a25-11e8-8ce9-a0a727ed006c.gif)

Notice, that the header highlight is not changing if the tooltip is shown.

**Comment by joachimmarder** [comment]
Possible reason for this behavior: Initially VCL creates a `THintWindow` instance and assigns it to `TApplication.FHintWindow`. `THintWindow.IsHintMsg()` tells the VCL to cancel the hint if a `WM_NCMOUSEMOVE` message arrive. `TVirtualTreeHintWindow.IsHintMsg()` does explicitely filter out those message regarding hint cancellation. This seems to be the reason that the header hint works as soon as a `TVirtualTreeHintWindow`was displayed and so `TApplication.FHintWindow` is an instance of this class.

Another possible reason: The function `FindVCLWindow()` in unit `Vcl.Controls` returns nil if the coordinates of the Virtual TreeView Header are passed, as this is in the non-client area of the control, so `TApplication.HintMouseMessage()` does not recognize that a hint should be shown for the header.

**Comment by gwideman** [comment]
I wonder if there is there any update (or subsequent issue) on this issue, because this problem appears quite significantly in 7.6.5 (May 2023), at least in the application I am using it in.  

I realize there's a V8.0.x generation, but I don't want to deal with breaking changes right now, and I didn't see any related items in the change logs.  Thanks.

In case it's any use, I could add my observations:

During a single run of the program, when I float the mouse over the column header, it may or may not show a hint. Once I get one header to show a hint, other headers will also show hints. Then at some point hints stop showing, and continue to stop showing for a while... but then start showing again.

I have some other normal VCL controls on the form, and their hints always show.

One of the things that often causes the VST column header hints to stop showing is activating one of the other controls' hints. 

And one action that often gets the VST column header to show again is selecting a row in the VST and scrolling it (scrollwheel).

So I can somewhat reliably create a sequence of  broken/works by alternating between activating the other controls' hints (VST hints stop working) then select and scroll a VST row, and hints start working again.

For what it's worth, this is in Delphi Athens, and VirtualStringTree is version 7.6.5
